lib/buffer.js

```diff
@@ -247,13 +248,13 @@ function fromArrayBuffer(obj, byteOffset, length) {
 }
 
 function fromObject(obj) {
-  if (obj instanceof Buffer) {
+  if (isUint8Array(obj)) {
     const b = allocate(obj.length);
 
     if (b.length === 0)
       return b;
 
-    obj.copy(b, 0, 0, obj.length);
+    Buffer.prototype.copy.call(obj, b, 0, 0, obj.length);
```

Comment:
Shouldn't need to reference it this way. `copy()` is native method. Can simply call `binding.copy(...)`.